File: src/lib/__tests__/end-screen.test.ts
import { describe, expect, it } from "vitest";
import fixture from "./fixtures/end-screen-stale-cases.json";
import { END_SCREEN_STALE_S, isEndScreenStale } from "../end-screen";
import { IDLE_RESET_MS } from "../idle-reset";

describe("isEndScreenStale (공유 fixture)", () => {
  for (const c of fixture.cases) {
    it(c.name, () => {
      expect(isEndScreenStale(c.secondsSinceEnd)).toBe(c.expect);
    });
  }

  it("NaN·무한은 소거하지 않는다", () => {
    expect(isEndScreenStale(NaN)).toBe(false);
    expect(isEndScreenStale(Infinity)).toBe(false);
  });

  it("앱 유휴 리셋(10분)보다 넉넉하다 — 잠깐 다른 앱을 보고 돌아오는 사용을 자르지 않는 하한", () => {
    expect(END_SCREEN_STALE_S * 1000).toBeGreaterThan(IDLE_RESET_MS);
  });
});

File: android/kit/src/test/kotlin/space/dodoplanet/gildongmu/kit/EndScreenTest.kt
package space.dodoplanet.gildongmu.kit

import kotlinx.serialization.Serializable
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertTrue

/** 웹 정본과 같은 공유 fixture(`end-screen-stale-cases.json`)를 읽는다 — Kit `EndScreenTests` 미러. */
class EndScreenTest {
    @Serializable
    private data class FixtureFile(val cases: List<Case>) {
        @Serializable
        data class Case(val name: String, val secondsSinceEnd: Double, val expect: Boolean)
    }

    @Test fun `종료 화면 수명 공유 fixture 동조`() {
        val cases = Fixtures.sharedJson("end-screen-stale-cases.json", FixtureFile.serializer()).cases
        assertTrue(cases.isNotEmpty())
        for (c in cases) assertEquals(c.expect, isEndScreenStale(c.secondsSinceEnd), c.name)
    }

    @Test fun `NaN·무한은 소거하지 않는다`() {
        assertFalse(isEndScreenStale(Double.NaN))
        assertFalse(isEndScreenStale(Double.POSITIVE_INFINITY))
    }
}
